### DISCUSSION TITLE
Tracking Scripts

### DISCUSSION THREAD

**Discussion by bigskillet** [swup/swup#879] [General]
Is it necessary to reload tracking scripts like Google Analytics, Facebook, Pinterest, etc?

I'm aware for the [GA Plugin](https://swup.js.org/plugins/ga-plugin), but it mentions that Google tracks the view event by default. Do I need to use the plugin if that's the case? ...and is the plugin compatible with the latest GA4 tracking script?

What about third-party social media tracking scripts like Facebook? Do they automatically track page views, or do they need to be reloaded somehow? I noticed there are no plugins for those.

I'm trying to wrap my head around what needs to be done exactly, thanks!

**Comment by daun** [comment]
That depends on the analytics service in question. If their tracking script is built to automatically react to url changes (sometimes called SPA mode or browser history mode), you can usually get away with using it exactly the way you would without swup. I think GA4 has an "enhanced measurements" mode that tracks browser history updates. Don't know about any of the other ones.

To keep things simple, I'd suggest using an abstraction like the [analytics package](https://www.npmjs.com/package/analytics) with plugins for all the different analytics providers. Then you'll only need to set things up once and you're free to add services without having to configure everything all over again.

**Reply by bigskillet** [reply]
Thanks @daun,

I'm building a simple static site and wanting to add transitions. The Google Analytics tracking would be very basic. Copy and paste from the analytics admin, nothing special. I just want to make sure it actually works and that page visits and views are recorded. The same goes for social media scripts. I just want to make sure everything is working properly. It sounds like most tracking scripts can detect SPA page views and the GA Plugin may be for special use cases?

Thanks for the tip about the analytics abstraction. I'll take a look, but it may be overkill in this case, and a little beyond my capabilities at this point.
